python/mxnet/ndarray.py

```diff
@@ -999,6 +999,85 @@ def concatenate(arrays, axis=0, always_copy=True):
 
     return ret
 
+# pylint: disable= no-member, protected-access, too-many-arguments
+def arange(start=None, stop=None, step=None, repeat=1, ctx=None, dtype=mx_real_t):
+    """Simlar function in the MXNet ndarray as numpy.arange
+        See Also https://docs.scipy.org/doc/numpy/reference/generated/numpy.arange.html.
+
+    Parameters
+    ----------
+    start : number, optional
+        Start of interval. The interval includes this value. The default start value is 0.
+    stop : number, optional
+        End of interval. The interval does not include this value.
+    step : number, optional
+        Spacing between values
+    repeat : number, optional
+        "The repeating time of all elements.
+        E.g repeat=3, the element a will be repeated three times --> a, a, a.
+    ctx : Context, optional
+        The context of the NDArray, default to current default context.
+    dtype : type, optional
+        The value type of the NDArray, default to np.float32
+    Returns
+    -------
+    out : NDArray
+        The created NDArray
+    """
+    if stop is None:
+        if start is None:
+            raise ValueError("Required argument \"start\"")
+        stop = start
+        start = 0
+    if step is None:
+        step = 1
+    if start is None:
+        start = 0
+    if ctx is None:
+        ctx = Context.default_ctx
+    return _internal._arange(start=start, stop=stop, step=step, repeat=repeat,
+                             dtype=_DTYPE_NP_TO_MX[np.dtype(dtype).type], ctx=str(ctx))
```

Comment:
@piiswrong I realize that if we include the `dtype` as one of the argument, we may need to use an additional wrapper in order to call  `_DTYPE_NP_TO_MX[np.dtype(dtype).type]`